src/main/java/uk/gov/moj/sdt/service/FeedbackService.java

```diff
@@ -0,0 +1,133 @@
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+import org.springframework.util.StringUtils;
+import uk.gov.moj.sdt.cmc.consumers.converter.XmlConverter;
+import uk.gov.moj.sdt.cmc.consumers.response.ClaimResponse;
+import uk.gov.moj.sdt.cmc.consumers.response.JudgementWarrantResponse;
+import uk.gov.moj.sdt.cmc.consumers.response.JudgmentWarrantStatus;
+import uk.gov.moj.sdt.cmc.consumers.response.ProcessingStatus;
+import uk.gov.moj.sdt.cmc.consumers.response.WarrantResponse;
+import uk.gov.moj.sdt.cmc.consumers.response.judgement.JudgementResponse;
+import uk.gov.moj.sdt.dao.api.IIndividualRequestDao;
+import uk.gov.moj.sdt.domain.ErrorLog;
+import uk.gov.moj.sdt.domain.api.IErrorLog;
+import uk.gov.moj.sdt.domain.api.IIndividualRequest;
+import uk.gov.moj.sdt.request.CMCFeedback;
+import uk.gov.moj.sdt.utils.cmc.RequestType;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+
+@Service
+@Slf4j
+public class FeedbackService {
+
+    private IIndividualRequestDao individualRequestDao;
+
+    private XmlConverter xmlToObject;
+
+    public FeedbackService(@Qualifier("IndividualRequestDao")
+                               IIndividualRequestDao individualRequestDao,
+                           XmlConverter xmlToObject) {
+        this.individualRequestDao = individualRequestDao;
+        this.xmlToObject = xmlToObject;
+    }
+
+    public void feedback(String sdtRequestId, CMCFeedback cmcFeedback) {
+        IIndividualRequest individualRequest = individualRequestDao.getRequestBySdtReference(sdtRequestId);
+        String requestType = individualRequest.getRequestType();
+        if (RequestType.JUDGMENT.getType().equals(requestType)) {
+            updateJudgementRequest(cmcFeedback, individualRequest);
+        } else if (RequestType.BREATHING_SPACE.getType().equals(requestType)) {
+            updateBreathingSpaceRequest(cmcFeedback, individualRequest);
+        } else if (RequestType.CLAIM_STATUS_UPDATE.getType().equals(requestType)) {
+            updateClaimStatusUpdateRequest(cmcFeedback, individualRequest);
+        } else if (RequestType.CLAIM.getType().equals(requestType)) {
+            updateCreateClaimRequest(cmcFeedback, individualRequest);
+        } else if (RequestType.WARRANT.getType().equals(requestType)) {
+            updateWarrantRequest(cmcFeedback, individualRequest);
+        } else if (RequestType.JUDGMENT_WARRANT.getType().equals(requestType)) {
+            updateCombinedJudgementWarrantRequest(cmcFeedback, individualRequest);
+        }
+        individualRequestDao.persist(individualRequest);
+    }
+
+    private void updateCreateClaimRequest(CMCFeedback cmcFeedback,
+                                          IIndividualRequest individualRequest) {
+        addStatusAndErrorLog(cmcFeedback, individualRequest);
+        ClaimResponse response = new ClaimResponse();
```

Comment:
Should this also set the fee field on the ClaimResponse?  The CMC Asynchronous Responses Confluence page mentions that this can be returned for a claim.

Reply:
done